src/link/riscv.zig

```diff
@@ -50,27 +50,27 @@ pub fn writeAddend(
 }
 
 pub fn writeInstU(code: *[4]u8, value: u32) void {
-    var data: Instruction = .{ .U = mem.bytesToValue(std.meta.TagPayload(Instruction, .U), code) };
+    var data: Instruction = .{ .U = mem.bytesToValue(@FieldType(Instruction, @tagName("U")), code) };
```

Comment:
Something went wrong here

```suggestion
    var data: Instruction = .{ .U = mem.bytesToValue(@FieldType(Instruction, "U"), code) };
```